File: app/src/main/cpp/project/CallbackTester.cpp
//
// Created by Naum Puroski on 24/02/2017.
//

#include <cwchar>
#include "CallbackTester.h"
#include "SimpleClass.h"

CallbackTester::CallbackTester(int value) : _value(value){

}

void CallbackTester::do_callback(Callback* callback) {
    if (callback) {
        callback->something_happened(_value);
    }
}
void CallbackTester::do_callback(std::function<void(int)> callback) {
    callback(_value);
}

File: android/app/src/main/java/com/nomenas/wrapperconcept/project/CallbackTester.java
package com.nomenas.wrapperconcept.project;

/**
 * Created by naum.puroski on 10/03/2017.
 */

public class CallbackTester extends Object {
    public CallbackTester(int value) {
        create(value);
    }

    public final native void do_callback(Callback callback);
    public final native void do_callback(LambdaCallback callback);

    protected CallbackTester() {};
    protected final native void create(int value);
    protected native void delete();
}
